### PR TITLE
Document the process for making point releases

### PR DESCRIPTION
**Pull request by ararslan**
This PR adds a large section in DISTRIBUTING.md to document in great detail how to make point releases. This is basically a more wordy version of the notes I was taking while making 0.6.1.

### CODE REVIEW

File: DISTRIBUTING.md [hunk 1]
```diff
@@ -169,3 +169,352 @@ binary packages. It also includes miscellaneous tools to do things such as
...
+
+## Checking for performance regressions
+
+Point releases should never introduce performance regressions.
+Luckily the Julia benchmarking bot, Nanosoldier, can run benchmarks against any
+branch, not just master.
+In this case we want to check the benchmark results of js/backport-x.y against
+release-x.y.
+To do this, we awaken the Nanosoldier from his robotic slumber using a comment
+on your backporting pull request:
```
Comment (waldyrious) [inline DISTRIBUTING.md]: There seems to be an inconsistency here: either it should read "we ... our" or "you ... your". Or perhaps better:

> To do this, awaken the Nanosoldier from his robotic slumber using a comment on the backporting pull request

File: DISTRIBUTING.md [hunk 2]
```diff
@@ -169,3 +169,352 @@ binary packages. It also includes miscellaneous tools to do things such as
...
+the Microsoft website.
+We need the `SignTool` utility which should have been installed somewhere like
+`C:\Program Files (x86)\Windows Kits\10\App Certification Kit`.
+Grab the Windows certificate files from CodeSigning on `juliasecure` and put them
+in the same directory as the executables.
+Open a Windows CMD window, `cd` to where all the files are, and run
+
+```
+set PATH=%PATH%;C:\Program Files (x86)\Windows Kits\10\App Certification Kit;
+signtool sign /f julia-windows-code-sign_2015.p12 /p "Juli@343" ^
```
Comment (tkelman) [inline DISTRIBUTING.md]: you shouldn't publicly post the password to a certificate

Comment (ararslan) [reply DISTRIBUTING.md]: ... 😐 whoops

**Comment by ararslan** [comment]
This is ready as far as I'm concerned. We can always add or change info as needed, or if we find that someone going through this process is still confused.

File: DISTRIBUTING.md (hunk 2 above)
Comment (StefanKarpinski) [reply DISTRIBUTING.md]: to state the obvious, that password should now be changed on the image as well as here.

Comment (ararslan) [reply DISTRIBUTING.md]: Yes, both have been changed.

**Pull request merged**